src/lib/sputter/net/port.cpp

```diff
@@ -0,0 +1,108 @@
+#include "port.h"
+
+#include <cstring>
+#include <iostream>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+using namespace sputter::net;
+
+UDPPort::UDPPort()
+    : socketHandle(-1) {
+}
+
+UDPPort::~UDPPort() {
+  close();
+}
+
+bool UDPPort::open(uint16_t port) {
+  socketHandle = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+  if (socketHandle < 0) {
+    std::cerr << "Error: Failed to create socket\n";
+    return false;
+  }
+
+  // Store port address
+  sockaddr_in bindAddress;
+  memset(&bindAddress, 0, sizeof(bindAddress));
+  bindAddress.sin_family = AF_INET;
+  bindAddress.sin_addr.s_addr = htonl(INADDR_ANY);
+  bindAddress.sin_port = htons(port);
+  if (bind(socketHandle, (sockaddr *)&bindAddress, sizeof(bindAddress)) < 0) {
+    std::cerr << "Error: Failed to bind socket\n";
```

Comment:
All of the error lines in port.cpp are good candidates for `LOG(WARNING)`

Reply:
Also if you could use C++ style casts that would be preferable!

i.e. `reinterpret_cast<sockaddr*>(&bindAddress)`

Reply:
236f023 and 65a8697